### ISSUE TITLE
Rendering issue in @open-wc/scoped-elements v3 for the Lion project

### ISSUE THREAD

**Issue by okadurin** [open-wc/open-wc#2766]
## Expected behavior

Components in [Lion](https://github.com/ing-bank/lion) project render correctly after updating `@open-wc/scoped-elements` from `v2` to `v3`.

## Actual Behavior

During updating `@open-wc/scoped-elements` from `v2` to `v3` at the `Lion` project, we discovered that some of UI components do not render anymore.
Particularly if the `lion-listbox` element [comes first](https://github.com/ing-bank/lion/blob/master/packages/ui/components/form-integrations/test/helpers/umbrella-form.js#L112) and then the `lion-select-rich` element [comes after it](https://github.com/ing-bank/lion/blob/master/packages/ui/components/form-integrations/test/helpers/umbrella-form.js#L130), then `lion-select-rich` does not render. When debugging it was discovered that `lion-select-rich` is using the same registry which `lion-listbox` has set when it was instantiated. The reason of the issue is in the new way `@open-wc/scoped-elements@v3` sets the registry for the elements. [Here](https://github.com/open-wc/open-wc/blob/scoped-elements-v2/packages/scoped-elements/src/ScopedElementsMixin.js#L117) is how it was done in v2 and [here](https://github.com/open-wc/open-wc/blob/%40open-wc/scoped-elements%403.0.3/packages/scoped-elements/html-element.js#L60) is how it is done for v3 . You could notice that `shouldCreateRegistry` contains more checks and it was reduced just to `!this.registry` which is not enough for our case.

Note if `lion-select-rich` comes first and then `lion-listbox` comes second, everything is rendered correctly.

### Code demo
I have prepared 2 PRs for providing more context. 

The [first one](https://github.com/ing-bank/lion/pull/2171/files) is the code from `Lion`'s master with `@open-wc/scoped-elements@v2` where the particular test file is focused and polyfill enabled. The tests in this PR do not produce any errors. 

The [second PR](https://github.com/ing-bank/lion/pull/2170/files) is done on top of the first one with the only difference that `@open-wc/scoped-elements` is update to `v3` and the minimum required code changed to address breaking changes introduced in `v3`.

To run the tests and see the related errors, one needs to run the steps as follows:
* Checkout the branch from the second PR: `git chekout bugfix/rich-select-with-polyfill-and-scoped-elements-v3`
* Remove `node_modules` just in case: `npm run rm-all-node_modules`
* `npm install`
* Run the test: `npx web-test-runner`

Then when running the tests here are the errors from the console:

```
🚧 Browser logs on Firefox:
      TypeError: this._invokerNode.retractRequestToBeDisabled is not a function
        at updated (packages/ui/components/select-rich/src/LionSelectRich.js:195:27)
        at _$AE (node_modules/@lit/reactive-element/src/reactive-element.ts:1376:9)
        at performUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1341:11)
        at scheduleUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1259:16)
        at _$Ej (node_modules/@lit/reactive-element/src/reactive-element.ts:1231:24)
        at async*requestUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1206:34)
        at requestUpdate (packages/ui/components/core/src/DisabledMixin.js:67:13)
        at requestUpdate (packages/ui/components/form-core/src/utils/SyncUpdatableMixin.js:102:13)
        at requestUpdate (packages/ui/components/form-core/src/InteractionStateMixin.js:42:13)
        at requestUpdate (packages/ui/components/overlays/src/OverlayMixin.js:66:13)
        at requestUpdate (packages/ui/components/select-rich/src/LionSelectRich.js:170:11)
        at u (node_modules/@lit/reactive-element/src/reactive-element.ts:944:9)
        at d (node_modules/@lit/reactive-element/src/reactive-element.ts:927:9)
        at s (node_modules/lit-element/src/lit-element.ts:115:0)
        at ScopedElementsHost (node_modules/@open-wc/scoped-elements/html-element.js:21:3)
        at ScopedElementsHost (node_modules/@open-wc/scoped-elements/lit-element.js:22:3)
        at SlotMixin (packages/ui/components/core/src/SlotMixin.js:39:7)
        at packages/ui/components/core/src/DisabledMixin.js:25:7
        at packages/ui/components/form-core/src/utils/SyncUpdatableMixin.js:30:7
        at packages/ui/components/form-core/src/registration/FormRegisteringMixin.js:24:7
        at FormControlMixin (packages/ui/components/form-core/src/FormControlMixin.js:157:7)
        at packages/ui/components/form-core/src/validate/ValidateMixin.js:135:7
        at InteractionStateMixin (packages/ui/components/form-core/src/InteractionStateMixin.js:59:7)
        at FocusMixin (packages/ui/components/form-core/src/FocusMixin.js:36:7)
        at packages/ui/components/form-core/src/registration/FormRegistrarMixin.js:38:7
        at ChoiceGroupMixin (packages/ui/components/form-core/src/choice-group/ChoiceGroupMixin.js:127:7)
        at ListboxMixin (packages/ui/components/listbox/src/ListboxMixin.js:234:7)
        at LionListbox (packages/ui/components/listbox/src/LionListbox.js:11:8)
        at OverlayMixin (packages/ui/components/overlays/src/OverlayMixin.js:28:7)
        at LionSelectRich (packages/ui/components/select-rich/src/LionSelectRich.js:110:5)
        at J (node_modules/@webcomponents/scoped-custom-element-registry/src/scoped-custom-element-registry.js:385:6)
        at c (node_modules/@webcomponents/scoped-custom-element-registry/src/scoped-custom-element-registry.js:236:10)
        at v (node_modules/lit-html/src/lit-html.ts:1122:51)
        at $ (node_modules/lit-html/src/lit-html.ts:1528:32)
        at _$AI (node_modules/lit-html/src/lit-html.ts:1375:11)
        at Z (node_modules/lit-html/src/lit-html.ts:2169:7)
        at update (node_modules/lit-element/src/lit-element.ts:165:23)
        at performUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1327:13)
        at scheduleUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1259:16)
        at _$Ej (node_modules/@lit/reactive-element/src/reactive-element.ts:1231:24)
        at async*requestUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1206:34)
        at u (node_modules/@lit/reactive-element/src/reactive-element.ts:944:9)
        at d (node_modules/@lit/reactive-element/src/reactive-element.ts:927:9)
        at s (node_modules/lit-element/src/lit-element.ts:115:0)
        at UmbrellaForm (packages/ui/components/form-integrations/test/helpers/umbrella-form.js:30:8)
        at J (node_modules/@webcomponents/scoped-custom-element-registry/src/scoped-custom-element-registry.js:385:6)
        at c (node_modules/@webcomponents/scoped-custom-element-registry/src/scoped-custom-element-registry.js:236:10)
        at v (node_modules/lit-html/src/lit-html.ts:1122:51)
        at $ (node_modules/lit-html/src/lit-html.ts:1528:32)
        at _$AI (node_modules/lit-html/src/lit-html.ts:1375:11)
        at Z (node_modules/lit-html/src/lit-html.ts:2169:7)
        at litFixtureSync (node_modules/@open-wc/testing-helpers/src/litFixture.js:32:9)
        at litFixture (node_modules/@open-wc/testing-helpers/src/litFixture.js:63:28)
        at fixture (node_modules/@open-wc/testing-helpers/src/fixture-no-side-effect.js:62:12)
        at packages/ui/components/form-integrations/test/dialog-integrations.test.js:21:48
 🚧 Browser logs on Chromium:
      TypeError: this._invokerNode.retractRequestToBeDisabled is not a function
        at LionSelectRich.updated (packages/ui/components/select-rich/src/LionSelectRich.js:195:27)
        at LionSelectRich._$AE (node_modules/@lit/reactive-element/src/reactive-element.ts:1376:9)
        at LionSelectRich.performUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1341:11)
        at LionSelectRich.scheduleUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1259:16)
        at LionSelectRich._$Ej (node_modules/@lit/reactive-element/src/reactive-element.ts:1231:24)
        at async LionSelectRich._$Ej (node_modules/@lit/reactive-element/src/reactive-element.ts:1220:4)
        at async LionSelectRich._alignInvokerWidth (packages/ui/components/select-rich/src/LionSelectRich.js:429:5)
 🚧 Browser logs on Webkit:
      TypeError: this._invokerNode.retractRequestToBeDisabled is not a function. (In 'this._invokerNode.retractRequestToBeDisabled()', 'this._invokerNode.retractRequestToBeDisabled' is undefined)
        at updated (packages/ui/components/select-rich/src/LionSelectRich.js:195:53)
        at _$AE (node_modules/@lit/reactive-element/src/reactive-element.ts:1376:17)
        at performUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1341:23)
        at node_modules/@lit/reactive-element/src/reactive-element.ts:1231:24
      Error: [OverlayController] Could not find a render target, since the provided contentNode is not connected to the DOM. Make sure that it is connected, e.g. by doing "document.body.appendChild(contentNode)", before passing it on.
        at rearrangeNodes (packages/ui/components/overlays/src/OverlayController.js:38:20)
        at __initContentDomStructure (packages/ui/components/overlays/src/OverlayController.js:596:19)
        at _init (packages/ui/components/overlays/src/OverlayController.js:509:37)
        at updateConfig (packages/ui/components/overlays/src/OverlayController.js:470:15)
        at OverlayController (packages/ui/components/overlays/src/OverlayController.js:197:22)
        at _defineOverlay (packages/ui/components/overlays/src/OverlayMixin.js:83:35)
        at _setupOverlayCtrl (packages/ui/components/overlays/src/OverlayMixin.js:222:46)
        at _setupOverlayCtrl (packages/ui/components/input-datepicker/src/LionInputDatepicker.js:311:28)
        at firstUpdated (packages/ui/components/overlays/src/OverlayMixin.js:171:29)
        at firstUpdated (packages/ui/components/input-datepicker/src/LionInputDatepicker.js:197:23)
        at _$AE (node_modules/@lit/reactive-element/src/reactive-element.ts:1374:24)
        at performUpdate (node_modules/@lit/reactive-element/src/reactive-element.ts:1341:23)
        at packages/ui/components/localize/src/LocalizeMixin.js:71:26
```
</details>

### Proposed fix
As the fix [the third PR](https://github.com/ing-bank/lion/pull/2172/files) was created. It is based on the PR 2. The only difference is the [patch](https://github.com/ing-bank/lion/pull/2172/files#diff-73e5e58315a514090fc15174b701adb323bb8c1530b53a1f16b956c5eb34e541) was added for `@open-wc/scoped-elements/html-element.js`. The patch is about bringing `shouldCreateRegistry` check from `v2` to `v3`.

What do you think about bringing that extra check back?

**Comment by thepassle** [comment]
It's not entirely clear to me why it works with the additional check. `this.registry` is a getter that returns `(this.constructor).__registry`

So for the following check:
```js
const shouldCreateRegistry =
        !this.registry ||
        // @ts-ignore
        (this.registry === this.constructor.__registry &&
          !Object.prototype.hasOwnProperty.call(this.constructor, '__registry'));
```

why do we need this?
```js
        (this.registry === this.constructor.__registry &&
          !Object.prototype.hasOwnProperty.call(this.constructor, '__registry'));
```

**Comment by okadurin** [comment]
> Are you overwriting the registry in Lion by any chance?

I have just checked the code and have not found any case where the `registry` is being overridden

> why do we need this?

Somehow whatever `lion-listbox` sets as its own `registry`, is available via prototype inheritance to [lion-select-rich](https://github.com/ing-bank/lion/blob/master/packages/ui/components/select-rich/src/LionSelectRich.js#L26).
Without that extra check `this.registry` is not `undefined` when `lion-select-rich` comes after `lion-listbox`.  In this case `this.registry` is the same object for both `lion-select-rich` and `lion-listbox` elements and this condition is `true`:
```
[$('lion-select-rich').registry === $('lion-listbox').registry
```
When `lion-select-rich` comes first and `lion-listbox` comes second, the above condition is false.

**Comment by okadurin** [comment]
Sure, let me try

**Comment by thepassle** [comment]
fixed in #2767
